src/core_modules/capture-core/components/Dialogs/DiscardDialog.constants.js

```diff
@@ -0,0 +1,9 @@
+// @flow
+import i18n from '@dhis2/d2-i18n';
+
+export const defaultDialogProps = {
+    header: i18n.t('Discard unsaved changes?'),
+    text: i18n.t('This event has unsaved changes. Leaving this page without saving will lose these changes. Are you sure you want to discard unsaved changes?'),
+    destructiveText: i18n.t('Yes, discard changes'),
```

Comment:
Same as mentioned in `DiscardDialog.component.js`:
If this dialog is meant to be used as both `Confirm` and `Discard`, then this default prop would still be passed in even if `confirmText` is set. The result would be having two buttons. (📸)

<img width="450" alt="image" src="https://user-images.githubusercontent.com/12842868/210763356-18c8c1ae-425f-45ea-84de-0131c35f9f51.png">

Reply:
hey @eirikhaugstulen, IMO we can leave it generic but it all depends on the design and use case.
I think of a case when we want to have multiple actions 1. No, stay here 2. Commit changes and navigate 3. Discard changes and navigate.
I think it's generally acceptable to have both primary and destructive button in the same scope. 
One of the example is in the app we have the action to Save/Cancel/Delete  
<img width="1440" alt="Screenshot 2023-01-05 at 11 18 25" src="https://user-images.githubusercontent.com/89806888/210768323-553a8300-c0c4-4b22-badd-733d0b5f403b.png">

We can discuss further in the team meeting if you want

Reply:
No, I definitely see your point in having three actions when you think of it like that 😅 The way I had pictured this widget was that you were only meant to ask the user if they really want to do the action they have clicked - not asking them which action to chose. But I'm not really sure of the use case here, so having three actions might be suitable.

@simonadomnisoru @JoakimSM any opinions?

Reply:
Hey, 

As of today, I'm not aware of any use case where both actions are needed in the same scope. Therefore, this premature optimization may never be necessary.
On the other hand, this pattern is already used in all forms and is familiar to the users. So if the use case occurs in the future, the component is ready to use.
I don't really have a strong opinion here. I see a good argument for both approaches and whichever path you choose is fine with me.

Thank you!

Reply:
If I am understanding it right, I also think just two actions are necessary. 1. Yes, discard changes (and move to wherever user clicked) and 2. No, cancel.

Adding the action "Yes, but save changes" will add too much complexity I think and the user can just click cancel and save in the traditional way.

Reply:
Hey @eirikhaugstulen,
I updated the code based on @karolinelien answer. The `Discard` button is the only one available and the modal props `destructiveText` and `onDestroy` have been changed to required. Let me know if you find other issues.
Thank you for the feedback! 😄